api/server/server.go

```diff
@@ -115,9 +115,9 @@ func pwd() string {
 	return strings.Replace(cwd, "\\", "/", -1)
 }
 
-func WithDBURL(dbURL string) ServerOption {
+func WithDBURL(ctx context.Context, dbURL string) ServerOption {
```

Comment:
this seems more natural to put on `ServerOption`, since it's a func that will be called later on initialization. the func that returns a ServerOption func will be called from a different context and probably shouldn't really do any work (the idea is to be lazy). both the `WithDBURL` and `WithLogURL` should probably call `.New` for their respective things inside of a `ServerOption` func instead of in the enclosing func that returns them, I guess these two were something of a mistake in that sense, would you mind fixing this since it's now an issue?

Reply:
it seems reasonable to make `ServerOption` slightly more flexible in 2 ways, with a signature of:

```go
type ServerOption func(context.Context, *Server) error
```

then we can just bubble up errors and log / return them in `server.New()` instead of relying on `ServerOption` calling `log.Fatal` to terminate the program on errors. 

not sure this matters too much, maybe it's easier (it is simpler) just keeping it like it is. hopefully people aren't doing very complex things in initialization anyway